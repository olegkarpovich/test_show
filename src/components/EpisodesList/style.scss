.ep-blocks{
  display: flex;
  flex-wrap: wrap;
  width: 100%;
  justify-content: space-between;

  .ep-block-header {
    color: #FFF;
    background: #337AB7;
    font-weight: bold;
    border: 1px solid #1a4a73;
    text-transform: uppercase;
  }

  .ep-block{
    display: flex;
    width: 100%;
    margin-bottom: 0px;
    align-items: stretch;
    flex-wrap: nowrap;
    text-align: left;

    .ep-season-number,
    .ep-date-time {
      display: none;
    }

    div {
      width: 40%;
      font-size: 15px;
      border: none;
      font-family: Verdana, sans-serif;
      border-bottom: 1px solid #337AB7;
      padding: 5px 1%;

      a {
        color: #337AB7;
        text-decoration: none;
        cursor: pointer;

        &:hover {
          text-decoration: underline;
        }
      }
    }
  }
}

@media(max-width: 568px) {
  .ep-block-header {
    text-transform: none !important;
  }

  .ep-block {
    div {
      font-size: 10px !important;
      white-space: nowrap !important;
      overflow: hidden !important;
      text-overflow: ellipsis !important;
    }

    .ep-score,
    .ep-number,
    .ep-season,
    .ep-date,
    .ep-time {
      display: none;
    }

    .ep-season-number {
      width: 20% !important;
    }

    .ep-season-number,
    .ep-date-time {
      display: inline-block !important;
    }

    .ep-name {
      flex-shrink: 0;
    }
  }
}