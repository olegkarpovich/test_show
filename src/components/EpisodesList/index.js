import React, {Component} from 'react'
import {Link} from 'react-router-dom';
import './style.scss';

function EpisodesList(props) {
    const {episodes} = props;
    return (
        <section className="ep-blocks">
            <header className="ep-block ep-block-header">
                <div className="ep-season-number">Number</div>
                <div className="ep-season">Season</div>
                <div className="ep-number">Number</div>
                <div className="ep-date">Date</div>
                <div className="ep-time">Time</div>
                <div className="ep-date-time">Date</div>
                <div className="ep-name">Name</div>
                <div className="ep-score">Score</div>
            </header>
            {episodes.map((episode, i) => (
                <article key={i} className="ep-block">
                    <div className="ep-season-number">{`s${episode.season}-e${episode.number}`}</div>
                    <div className="ep-season">{episode.season}</div>
                    <div className="ep-number">{episode.number}</div>
                    <div className="ep-date">{episode.airdate}</div>
                    <div className="ep-time">{episode.airtime}</div>
                    <div className="ep-date-time">{`${episode.airdate} ${episode.airtime}`}</div>
                    <div className="ep-name"><Link to={`/season/${episode.season}/episode/${episode.number}`} > {episode.name}</Link> </div>
                    <div className="ep-score">-</div>
                </article>
            ))}
        </section>
    )
}
export default EpisodesList;
