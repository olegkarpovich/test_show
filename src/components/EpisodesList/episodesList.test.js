import React from 'react';
import { render, act } from '@testing-library/react';
import { BrowserRouter } from "react-router-dom";
import EpisodesList from './index';

const episodes = [
    {
        name: 'Escape from Monster Island',
        airdate: '2016-04-04',
        airtime: '18:00',
        number: 1,
        season: 1
    }
];

const component = () => (
    <BrowserRouter>
        <EpisodesList episodes={episodes}/>
    </BrowserRouter>
);

describe('EpisodesList component', () => {
    it('should render episodes list', () => {
        const { container } = render(component());

        expect(container).toMatchSnapshot();
    });
});