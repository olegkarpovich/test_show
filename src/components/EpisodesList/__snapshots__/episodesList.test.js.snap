// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodesList component should render episode details page 1`] = `
<div>
  <section
    class="ep-blocks"
  >
    <header
      class="ep-block ep-block-header"
    >
      <div
        class="ep-season-number"
      >
        Number
      </div>
      <div
        class="ep-season"
      >
        Season
      </div>
      <div
        class="ep-number"
      >
        Number
      </div>
      <div
        class="ep-date"
      >
        Date
      </div>
      <div
        class="ep-time"
      >
        Time
      </div>
      <div
        class="ep-date-time"
      >
        Date
      </div>
      <div
        class="ep-name"
      >
        Name
      </div>
      <div
        class="ep-score"
      >
        Score
      </div>
    </header>
    <article
      class="ep-block"
    >
      <div
        class="ep-season-number"
      >
        s1-e1
      </div>
      <div
        class="ep-season"
      >
        1
      </div>
      <div
        class="ep-number"
      >
        1
      </div>
      <div
        class="ep-date"
      >
        2016-04-04
      </div>
      <div
        class="ep-time"
      >
        18:00
      </div>
      <div
        class="ep-date-time"
      >
        2016-04-04 18:00
      </div>
      <div
        class="ep-name"
      >
        <a
          href="/season/1/episode/1"
        >
           
          Escape from Monster Island
        </a>
         
      </div>
      <div
        class="ep-score"
      >
        -
      </div>
    </article>
  </section>
</div>
`;

exports[`EpisodesList component should render episodes list 1`] = `
<div>
  <section
    class="ep-blocks"
  >
    <header
      class="ep-block ep-block-header"
    >
      <div
        class="ep-season-number"
      >
        Number
      </div>
      <div
        class="ep-season"
      >
        Season
      </div>
      <div
        class="ep-number"
      >
        Number
      </div>
      <div
        class="ep-date"
      >
        Date
      </div>
      <div
        class="ep-time"
      >
        Time
      </div>
      <div
        class="ep-date-time"
      >
        Date
      </div>
      <div
        class="ep-name"
      >
        Name
      </div>
      <div
        class="ep-score"
      >
        Score
      </div>
    </header>
    <article
      class="ep-block"
    >
      <div
        class="ep-season-number"
      >
        s1-e1
      </div>
      <div
        class="ep-season"
      >
        1
      </div>
      <div
        class="ep-number"
      >
        1
      </div>
      <div
        class="ep-date"
      >
        2016-04-04
      </div>
      <div
        class="ep-time"
      >
        18:00
      </div>
      <div
        class="ep-date-time"
      >
        2016-04-04 18:00
      </div>
      <div
        class="ep-name"
      >
        <a
          href="/season/1/episode/1"
        >
           
          Escape from Monster Island
        </a>
         
      </div>
      <div
        class="ep-score"
      >
        -
      </div>
    </article>
  </section>
</div>
`;
